pkg/router/controller/unique_host.go

```diff
@@ -119,11 +141,14 @@ func (p *UniqueHost) HandleRoute(eventType watch.EventType, route *routeapi.Rout
 		} else {
 			if oldest.CreationTimestamp.Before(route.CreationTimestamp) {
 				glog.V(4).Infof("Route %s cannot take %s from %s", routeName, host, routeNameKey(oldest))
-				return fmt.Errorf("route %s holds %s and is older than %s", routeNameKey(oldest), host, key)
+				err := fmt.Errorf("another route holds %s and is older than %s", host, route.Name)
```

Comment:
Would it make sense for this error message to be explicit that the other route is in a different namespace?  It's clear to a programmer who can see where the error is thrown, but a user might not have that advantage.